### PR TITLE
Bga update

### PR DESCRIPTION
**Pull request by yankee14**

### CODE REVIEW

**Comment by yankee14** [comment]
[Associated symbol PR](https://github.com/KiCad/kicad-symbols/pull/1866)
[Associated footprint PR](https://github.com/KiCad/kicad-footprints/pull/1604)

File: scripts/Packages/Package_BGA/bga.yaml [hunk 1]
```diff
@@ -1,114 +1,140 @@
...
+  body_size_y: 10.0
   pitch: 0.8
   pad_diameter: 0.4
   mask_margin: 0.035
   paste_margin: 0.000001
   layout_x: 11
   layout_y: 11
+TFBGA-196_11.0x11.0mm_Layout14x14_P0.75mm_Ball0.32mm_Pad0.35mm_SMD:
+  description: "TFBGA-196"
+  size_source: "http://ww1.microchip.com/downloads/en/DeviceDoc/SAMA5D2-Series-Data-Sheet-DS60001476C.pdf"
```
Comment (evanshultz) [inline scripts/Packages/Package_BGA/bga.yaml]: Add "#page=2956" onto the end of the link to just right to this package's page

File: scripts/Packages/Package_BGA/bga.yaml [hunk 2]
```diff
@@ -1,114 +1,140 @@
...
+  size_source: "http://ww1.microchip.com/downloads/en/PackagingSpec/00000049BQ.pdf#p495"
+  body_size_x: 10.0
+  body_size_y: 10.0
   pitch: 0.8
   pad_diameter: 0.4
   mask_margin: 0.035
   paste_margin: 0.000001
   layout_x: 11
   layout_y: 11
+TFBGA-196_11.0x11.0mm_Layout14x14_P0.75mm_Ball0.32mm_Pad0.35mm_SMD:
```
Comment (evanshultz) [inline scripts/Packages/Package_BGA/bga.yaml]: The ball diameter does not need to be specified, I don't think. What was the reason to add it?

**Comment by evanshultz** [comment]
I made a couple comments inline. Also, would you please add the package height of 1.2mm (see other entries for an example of the syntax). Can you also merge master into this PR if it's not?

**Comment by evanshultz** [comment]
And check your padstacks: I believe the margin is by radius so it gets doubled and thus your mask openings are shrunk twice the value they should be.
image

**Comment by yankee14** [comment]
I have no idea what's going on. I guess I'll create a new pull request.

**Comment by evanshultz** [comment]
Can you advise what the problem was, if in fact you're looking for help? There were just a few things above and then I thought this would be close to getting merged. Let us know if we can assist you.

**Comment by yankee14** [comment]
@evanshultz git and GitHub just surprised me is all; I don't understand what happened with my commits, or how mine/your pull requests were organized.

If I remember right, I needed to add my footprint to the BGA generator yaml file, but that file did not yet exist in kicad-footprint-generator master branch. You had a pull request called bga-update that contained the yaml file I had to make my entry in, so I think I made a pull request of your pull request. I had two remotes in my local repository:

1. origin - my fork
2. upstream - some remote repo... I don't remember what it said, but i know it had 'evanschultz' in the URL, which means it wasn't a pure upstream URL to kicad-footprint-generator. I think this upstream was your pull request, which was already merged and closed.

I saw that you asked:

> Can you merge master into this PR if it's not?

So I edited my upstream URL to point to kicad-footprint-generator, and fast-forward merged master into bga-update. I tried to push the result to origin, but it wouldn't let me. I figured GitHub wouldn't let me destroy someone else's git history, so I force pushed. What you see above is the result of that. Sorry, I don't know enough about git to explain what happened! Not sure if my PR was a unique PR off kicad-footprint-generator, or if it was a PR of your bga-update PR, or if we were pushing to the same PR.

**Pull request closed without merging**